### PR TITLE
feat: documentation updates for 0.4.0 enhancements

### PR DESCRIPTION
**Pull request by KellyMerrick**

### CODE REVIEW

File: content/usage/pipeline/steps/ruleset.md [hunk 1]
```diff
@@ -77,6 +77,43 @@ ruleset:
...
+    image: alpine:latest
+    commands:
+      - echo "i will never run with comment event"
+    ruleset:
+      event: [ push, pull_request ]
+
+  - name: echo_only_run_comment
+    image: alpine:latest
+    commands:
+      - echo "i will only run with comment event"
```
Comment (kneal) [inline content/usage/pipeline/steps/ruleset.md]: Wouldn't this step only run with comment event and a "run build" comment?

Comment (KellyMerrick) [reply content/usage/pipeline/steps/ruleset.md]: it has a run build comment on line 114, but i can make it clearer

**Review by kneal — APPROVED** [review]
LGTM 🐬

**Review by jbrockopp — APPROVED** [review]
LGTM

**Pull request merged**